vm/src/codecs.rs

```diff
@@ -623,7 +623,7 @@ fn surrogatepass_errors(err: PyObjectRef, vm: &VirtualMachine) -> PyResult<(PyOb
         // Try decoding a single surrogate character. If there are more,
         // let the codec call us again.
         let p = &s.as_bytes()[range.start..];
-        if p.len() - range.start >= byte_length {
+        if (p.len() as isize) - range.start as isize >= byte_length as isize {
```

Comment:
```suggestion
        if p.len() >= range.start + byte_length {
```
How about this? is this also panic?

Reply:
isize should not overflow downwards, I tested it and it fixed the issue.

Reply:
isize casting is usually not an issue on 64bit architecture but sometimes on 32bit one by limiting it half size. We can avoid it when possible. I will take current changes if removing subtraction doesn't fix problem.

Reply:
I switched to saturating subtraction, hopefully that works as well.